Realm/RLMObject.mm

```diff
@@ -40,7 +40,11 @@ - (instancetype)init {
 }
 
 - (instancetype)initWithValue:(id)value {
-    return [super initWithValue:value schema:RLMSchema.sharedSchema];
+    RLMSchema *schema = RLMSchema.partialSharedSchema;
+    if (![schema schemaForClassName:self.class.className]) {
```

Comment:
Just checking if the current class is contained in the partial schema isn't sufficient to ensure its linked properties' classes are already included.

This is why it seems more reasonable to me to replace `[RLMSchema sharedSchema]` with `[[RLMConfiguration defaultConfiguration] schema]` and to require that all `RLMObject` classes that will be used be included in the default configuration's schema.